packages/framework/src/utils/stats/timeSeries.ts

```diff
@@ -105,6 +105,22 @@ export class TimeSeriesStats<I, O> {
     for (const [timeFrameIndex, timeFrame] of sortedTimeFrames.entries()) {
       const timeFrameName = TimeFrame[timeFrame]
       console.log(`📈 processing ${type} ${timeFrameName} for ${account}`)
+
+      // @note: get the previous time frame, which is able to cleanly divide the current time frame.
+      let trueDivideTimeFrameIndex = timeFrameIndex
+      if (timeFrameIndex > 0) {
+        trueDivideTimeFrameIndex--
+      }
+      if (timeFrame !== TimeFrame.All && trueDivideTimeFrameIndex < 0) {
```

Comment:
shouldn't it be "&& trueDivideTimeFrameIndex > 0"?

Reply:
```suggestion
      if (timeFrame !== TimeFrame.All && trueDivideTimeFrameIndex > 0) {
```
uhm, you're right :fearful: